db/crud.go

```diff
@@ -297,19 +297,24 @@ func (db *Database) GetRevWithHistory(docid, revid string, maxHistory int, histo
 		body["_revisions"] = revisions
 	}
 
-	if showExp {
-		// If doc is nil (we got the rev from the rev cache), look up the doc to find the exp on the doc
-		if doc == nil {
-			if doc, err = db.GetDocument(docid, DocUnmarshalSync); doc == nil {
-				return nil, err
-			}
-		}
-		if doc.Expiry != nil && !doc.Expiry.IsZero() {
-			body["_exp"] = doc.Expiry.Format(time.RFC3339)
+	// If doc is nil (we got the rev from the rev cache)
+	// look up the doc sync to get attachment metadata and exp
```

Comment:
Add a TODO here about enhancing the rev cache to include attachments metadata (and split out that work into a separate github issue)

Reply:
Added comment and split out into #3688